tree.go

```diff
@@ -567,53 +566,81 @@ func (n *InternalNode) Commitment() *Point {
 	return n.commitment
 }
 
+func (n *InternalNode) fillLevels(levels [][]*InternalNode) {
+	levels[int(n.depth)] = append(levels[int(n.depth)], n)
+	for _, child := range n.children {
```

Comment:
I'm missing the point here: why not iterate over `n.cow` instead of all the children?

Reply:
There's no point actually, it will save some cycles, but doubt it will make a difference. In any case, is a good idea.
I changed it to use `cow` now (last commit).